addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1348,27 +1348,62 @@ const Wysiwyg = Widget.extend({
      * @param {Node} [params.htmlClass] Optionnal
      */
     openMediaDialog(params = {}) {
-        const sel = this.odooEditor.document.getSelection();
-
-        if (!sel.rangeCount) {
-            return;
-        }
-        const range = sel.getRangeAt(0);
         // We lose the current selection inside the content editable when we
         // click the media dialog button so we need to be able to restore the
         // selection when the modal is closed.
         const restoreSelection = preserveCursor(this.odooEditor.document);
 
-        const $editable = $(OdooEditorLib.closestElement(range.startContainer, '.o_editable') || this.odooEditor.editable);
-        const model = $editable.data('oe-model');
-        const field = $editable.data('oe-field');
-        const type = $editable.data('oe-type');
+        /**
+         * Looks up the attachment domain of the given element by traversing up the DOM tree
+         * until it finds an element with the necessary data attributes. If no such element is found,
+         * it returns the attachment domain of the editable element.
+         *
+         * @param {jQuery} $el - The jQuery element to start the search from.
+         * @returns {Object} An object containing the attachment domain information.
+         * @returns {string} resModel - The model name of the attachment.
+         * @returns {number|false} resId - The ID of the attachment, or false if it doesn't exist.
+         * @returns {Array} domain - The domain of the attachment.
+         * @returns {string|false} type - The type of the attachment, or false if it doesn't exist.
+         * @returns {string|false} field - The field of the attachment, or false if it doesn't exist.
+         */
+        function lookupAttachementDomain(el, top)  {
+            function extractAttachementDomain($el){
+                return {
+                    resModel: $el.data('oe-model') || 'ir.attachment',
+                    resId: $el.data('oe-id') || false,
+                    domain: $el.data('oe-media-domain') || [],
+                    type: $el.data('oe-type') || false,
+                    field : $el.data('oe-field') || false,
+                };
+            }
+
+            el = el.closest('.o_editable');
+
+            while (el) {
+                const $el = $(el);
+                if (($el.data('oe-model') && $el.data('oe-id')) || 
+                    $el.data('oe-media-domain')) {
+                    return extractAttachementDomain($el)
+                } 
+
+                if ($el.is(top)) {
+                    break;
+                }
+
+                el = el.parentNode?.closest('.o_editable') ?? null;
+            }
+
+            return extractAttachementDomain($(top));
+        }
+
+        const el = params.node ? params.node : this.odooEditor.document.getSelection().anchorNode;
```

Comment:
I still think this problem should not exist because the "Replace" button should have a `preventDefault` to avoid stealing the selection at all.

Also, I unresolved the other comment because it was not resolved.

Reply:
No, it's just not the button finally. Also, I don't like preventing the default this might have some bad implications for accessibility and doesn't solve the problem for people who are using keyboard shortcuts for selecting the button

Reply:
How to you manage to select the Replace button by using keyboard shortcuts ? I didn't find a way to do it. What do you mean "it's just not the button finally" ? Do you have another use case where the same bug happens ? What are the steps to reproduce ?

Reply:
> How to you manage to select the Replace button by using keyboard shortcuts ?

Using tab, but also if I select another widget in the UI, like the description field before clicking the button

> I didn't find a way to do it. What do you mean "it's just not the button finally" ?

Dunno, but there are other calls to this method that provide a param.node, and where the selection state might be unreliable, I think it's just safer to start at the param.node in general. I just kept the selection logic for the case we don't get a node as a param

Reply:
Ok I get it. I'm going to resolve this comment, but I'm still interested in finding a way to fix the issue that doesn't involve explicit looping on nodes and two level of inner functions, so I'm keeping the other one open.